File: packages/modules/status-bar/ios/StatusBar.m
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import "StatusBar.h"
#import <Foundation/Foundation.h>

@implementation StatusBar

+ (NSString *)name {
 return @"StatusBar";
}

+ (NSDictionary<NSString *, NSString *> *)methodLookup {
 return @{
  @"setStyle" : NSStringFromSelector(@selector(setStyleWithStyle:(NSString *)style)),
  @"setHidden" : NSStringFromSelector(@selector(setHiddenWithHidden:(BOOL)hidden)),
  @"setNetworkActivityIndicatorVisible" : NSStringFromSelector(@selector(setNetworkActivityIndicatorVisibleWithVisible:(BOOL)visible)),
  @"setBackgroundColor" : NSStringFromSelector(@selector(setBackgroundColorWithColor:(NSString *)color)),
 };
}

- (void)setStyleWithStyle:(NSString *)style {
}

- (void)setHiddenWithHidden:(BOOL)hidden {
}

- (void)setNetworkActivityIndicatorVisibleWithVisible:(BOOL)visible {
}

- (void)setBackgroundColorWithColor:(NSString *)color {
}

@end

File: packages/modules/status-bar/android/src/main/java/com/lynxpo/statusbar/StatusBar.java
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
package com.lynx.explorer.modules;

import android.content.Context;
import com.lynx.jsbridge.LynxMethod;
import com.lynx.jsbridge.LynxModule;
import com.lynx.jsbridge.LynxNativeModule;
import com.lynxpo.statusbar.generated.StatusBarSpec;
import com.lynx.react.bridge.WritableMap;
import com.lynx.react.bridge.JavaOnlyMap;
import com.lynx.react.bridge.WritableArray;
import com.lynx.react.bridge.JavaOnlyArray;

/**
 * Android counterpart of the iOS {@code StatusBar}. Exposes functionality to JS via
 * {@code NativeModules.StatusBar}, faithfully porting Expo's native method surface.
 */
@LynxNativeModule(name = "StatusBar")
public class StatusBar extends StatusBarSpec {

  public StatusBar(Context context) {
    super(context);
  }

  @LynxMethod
  public void setStyle(String style) {
  }

  @LynxMethod
  public void setHidden(Boolean hidden) {
  }

  @LynxMethod
  public void setNetworkActivityIndicatorVisible(Boolean visible) {
  }

  @LynxMethod
  public void setBackgroundColor(String color) {
  }

}
